test/tasks/dump.test.ts

```diff
@@ -0,0 +1,627 @@
+  const wrappedNativeToken = (undefined as unknown) as string;
+
+  let deployer: Wallet;
+  let user: Wallet;
+  let api = new MockApi();
+
+  let defaultDumpInstructions: Omit<
+    GetDumpInstructionInput,
+    "dumpedTokens" | "toTokenAddress"
+  >;
+
+  beforeEach(async () => {
+    consoleWarn = console.warn;
+    console.warn = (...args: unknown[]) => (consoleWarnOutput = args[0]);
+
+    [deployer, user] = await waffle.provider.getWallets();
+    api = new MockApi();
+    defaultDumpInstructions = {
+      user: user.address,
+      allowanceManager,
+      maxFeePercent: 100,
+      hasCustomReceiver: false,
+      hre,
+      network,
+      api,
+    };
+  });
+
+  afterEach(() => {
+    api.mock.assertAllExpectationsUsed();
+    console.warn = consoleWarn;
+    consoleWarnOutput = undefined;
+  });
+
+  it("dumps token for token", async () => {
+    const to = await mockErc20(deployer);
+    await to.mock.symbol.returns("TOTOKEN");
+    await to.mock.decimals.returns(101);
+    const dumped = await mockErc20(deployer);
+    await dumped.mock.symbol.returns("DUMPEDTOKEN");
+    await dumped.mock.decimals.returns(0xd);
+
+    const balance = utils.parseEther("42");
+    const fee = utils.parseEther("1");
+    const allowance = utils.parseEther("31337");
+    const boughtAmount = utils.parseEther("0.1337");
+
+    await dumped.mock.balanceOf.withArgs(user.address).returns(balance);
+    await dumped.mock.allowance
+      .withArgs(user.address, allowanceManager)
+      .returns(allowance);
+    mockApiCalls({
+      api,
+      toToken: to.address,
+      dumpedToken: dumped.address,
+      balance,
+      fee,
+      boughtAmount,
+    });
+
+    const {
+      toToken,
+      transferToReceiver,
+      instructions,
+    } = await getDumpInstructions({
+      ...defaultDumpInstructions,
+      dumpedTokens: [dumped.address],
+      toTokenAddress: to.address,
+    });
+
+    expect(toToken.symbol).to.deep.equal("TOTOKEN");
+    expect(toToken.decimals).to.deep.equal(101);
+    expect(isNativeToken(toToken)).to.be.false;
+    expect((toToken as Erc20Token).address).to.deep.equal(to.address);
+
+    expect(transferToReceiver).to.be.undefined;
+
+    expect(instructions).to.have.length(1);
+    const {
+      token: dumpedToken,
+      amountWithoutFee,
+      needsAllowance,
+      receivedAmount,
+      balance: returnedBalance,
+      fee: returnedFee,
+    } = instructions[0];
+    expect(dumpedToken.symbol).to.deep.equal("DUMPEDTOKEN");
+    expect(dumpedToken.decimals).to.deep.equal(0xd);
+    expect(dumpedToken.address).to.deep.equal(dumped.address);
+    expect(needsAllowance).to.deep.equal(false);
+    expect(receivedAmount).to.deep.equal(boughtAmount);
+    expect(amountWithoutFee).to.deep.equal(balance.sub(fee));
+    expect(returnedBalance).to.deep.equal(balance);
+    expect(returnedFee).to.deep.equal(fee);
+  });
+
+  for (const to of [undefined, BUY_ETH_ADDRESS]) {
+    it(`dumps token for eth (${to})`, async () => {
```

Comment:
The way I read this is "dumps token for eth (undefined)" and "dumps token for eth (0xeEeE...eE)"... is that what we wanted? Specifically the `undefined`.

Reply:
Yes, that's because the user can either omit the parameter (so `undefined`), which defaults to ETH, or explicitly choose ETH.

Reply:
I changed the test description to make it clear that `undefined` is an expected option.